File: Binary_Subarrays_With_Sum/submission3.cpp
class Solution {
public:
    int numSubarraysWithSum(vector<int>& nums, int goal) {
        
        int n = nums.size(), ans = 0, sum = 0;
        unordered_map<int,int> occurance;

        for (int i=0; i<n; i++){

            sum = sum + nums[i];

            if (sum==goal){
                ans++;
            }

            ans = ans + occurance[sum-goal];
            occurance[sum]++;
        }

        return ans;
    }
};

File: Maximum_Subarray/submission3.py
class Solution(object):
    def maxSubArray(self, nums):
        
        sum = nums[0]
        maxi = nums[0]
        i = 1

        while i<len(nums):
            if sum<0:
                sum = 0
            sum += nums[i]
            maxi = max(sum,maxi)
            i += 1
        
        return maxi